is_core/utils/__init__.py

```diff
@@ -139,87 +139,114 @@ def get_model_method_or_property_data(field_name, model, fun_kwargs):
     from is_core.forms.widgets import ReadonlyWidget, ManyToManyReadonlyWidget, ModelObjectReadonlyWidget
 
+    label = _get_model_field_label(field)
+
     if field.auto_created and (field.one_to_many or field.many_to_many):
         return (
-             None if isinstance(model, type) else [obj for obj in getattr(model, field.name).all()], (
-                getattr(field.field, 'reverse_verbose_name', None)
-                if getattr(field.field, 'reverse_verbose_name', None) is not None
-                else field.related_model._meta.verbose_name_plural
-            ), ManyToManyReadonlyWidget
+            None if isinstance(model, type) else [obj for obj in getattr(model, field.name).all()],
+            label,
+            ManyToManyReadonlyWidget
         )
     elif field.auto_created and field.one_to_one:
         return (
             None if isinstance(model, type) or not hasattr(model, field.name)
-            else getattr(model, field.name), (
-                getattr(field.field, 'reverse_verbose_name', None)
-                if getattr(field.field, 'reverse_verbose_name', None) is not None
-                else field.related_model._meta.verbose_name
-            ),  ModelObjectReadonlyWidget
+            else getattr(model, field.name),
+            label,
+            ModelObjectReadonlyWidget
         )
     elif field.many_to_many:
         return (
             None if isinstance(model, type) else [obj for obj in getattr(model, field.name).all()],
-            field.verbose_name, ManyToManyReadonlyWidget
+            label,
+            ManyToManyReadonlyWidget
         )
     else:
         return (
-            None if isinstance(model, type)
-            else _val_to_readonly_value(
+            None if isinstance(model, type) else _val_to_readonly_value(
                 getattr(model, field.name) if hasattr(model, field.name) else None, field, model
-            ), getattr(field, 'verbose_name', pretty_name(field.name)), ReadonlyWidget
+            ),
+            label,
+            ReadonlyWidget
         )
 
 
-def _get_model_readonly_data(field_name, model, fun_kwargs):
+def _get_model_readonly_data(field_name, model, fun_kwargs, field_labels=None):
+    field_labels = {} if field_labels is None else field_labels
+
     if '__' in field_name:
         current_field_name, next_field_name = field_name.split('__', 1)
         field = get_field_from_model_or_none(model, current_field_name)
-        if hasattr(model, current_field_name) and getattr(model, current_field_name):
-            return _get_model_readonly_data(next_field_name, getattr(model, current_field_name), fun_kwargs)
-        elif model and field and hasattr(field, 'related_model'):
-            return _get_model_readonly_data(next_field_name, field.related_model, fun_kwargs)
-        else:
+        if not field:
+            return None
+
+        next_value = getattr(model, current_field_name, None) if hasattr(model, current_field_name) else None
```

Comment:
`getattr(model, current_field_name, None)` is enough

Reply:
no it isnt. Because getattr(model, current_field_name, None) shoud raise ObjectDoesNotExist for o2o field. This is django fail..